### PR TITLE
Created helm chart for BaGetter

### PR DESCRIPTION
**Pull request by Khaos66**
Adds a helm chart to deploy all resources for BaGetter to k8s

Adresses #116

### CODE REVIEW

File: chart/bagetter/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "1.0.4"
```
Comment (seriouz) [inline chart/bagetter/Chart.yaml:15]: What about increasing this version number? We use the Git tags and GitHub actions to set the version of BaGetter. I think you have to edit the `.github/workflows/release.yml` and add a new job where the source code of the file (version number) gets replaced by the action.

File: chart/bagetter/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# bagetter
+
+![Version: 0.1.0](https://img.shields.io/badge/Version-0.1.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 1.0.4](https://img.shields.io/badge/AppVersion-1.0.4-informational?style=flat-square)
```
Comment (seriouz) [inline chart/bagetter/README.md:3]: AppVersion is here static, too.

**Comment by Regenhardt** [comment]
How about a `deployment templates` directory with its own readme, then in there a `Helm` directory for the chart, and a `Ansible-Helm` directory for the ansible playbook deploying the helm chart? That way others like docker-compose already have a set structure to be added to in the future and we have a readme there that can explain how to use the templates.

**Comment by seriouz** [comment]
Yeah then you could add Docker-Swarm too.

File: chart/bagetter/Chart.yaml (hunk 1 above)
Comment (Khaos66) [reply chart/bagetter/Chart.yaml:15]: Yes. I was thinking about that, too. 
I need to look into it...

File: chart/bagetter/README.md (hunk 2 above)
Comment (Khaos66) [reply chart/bagetter/README.md:3]: This file was generated by `helm-docs`
Should be re-generated by `release.yml`, too

**Comment by Khaos66** [comment]
I can move the directory to `deployment templates` sure.

**Comment by Khaos66** [comment]
@Regenhardt Thank you for the reminder!
I did move the chart.
I also extended `release.yml` with a job to update the chart.

**Review by Regenhardt — APPROVED** [review]

**Comment by HaimKortovich** [comment]
Hi, where is the helm chart hosted? there are no instructions to install it

**Comment by Regenhardt** [comment]
> Hi, where is the helm chart hosted? there are no instructions to install it

I was under the impression you don't have to "install" the helm chart but download the files, fill the variables with your values, and then execute it via helm. @Khaos66 you know more about this stuff, is there another step we should take here?

**Comment by Khaos66** [comment]
Yes, right. There is the concept of a help chart repo, where you can publish your charts to and users can install them from there. But this project doesn't have a repo. 

@HaimKortovich The way I use the chart is:
- git clone the repo 
- build the helm chart
- install the helm chart

```
helm dependency build ./bagetter/chart/bagetter

helm install -f myvalues.yml bagetter ./bagetter/chart/bagetter
```

See [helm install](https://helm.sh/docs/helm/helm_install/) for more info

**Comment by HaimKortovich** [comment]
A chart repo can be a github pages if its possible I can create the CI

**Comment by Amith211** [comment]
Just discovered this cool project. I would really appreciate a helm repo being setup to make my deployment easier as I'm using GitOps (specifically flux) for my k8s cluster.

**Comment by HaimKortovich** [comment]
Same here I'll try making a PR in the following hours

**Comment by seriouz** [comment]
@robford123 @Regenhardt I think we should create an extra project for the Helm Chart GitHub Pages. I don't like the idea of using the bagetter/BaGetter GitHub pages for the Helm releases.

**Pull request merged**